src/observers/CsharpLoggerObserver.ts

```diff
@@ -58,9 +58,23 @@ export class CsharpLoggerObserver extends BaseLoggerObserver {
             case Event.LatestBuildDownloadStart.name:
                 this.logger.appendLine("Getting latest OmniSharp version information");
                 break;
+            case Event.DownloadRetry.name:
+                this.handleDownloadRetry(<Event.DownloadRetry>event);
+                break;
+            case Event.CorruptedDownloadError.name:
+                this.handleCorruptedDownloadError(<Event.CorruptedDownloadError>event);
+                break;
         }
     }
 
+    private handleCorruptedDownloadError(event: Event.CorruptedDownloadError){
+        this.logger.appendLine(`There was a problem downloading ${event.packageDescription}. Some functionalities may not work as expected. Please restart vscode to retrigger the download or download the package manually from ${event.url}`); 
```

Comment:
functionalities -> features
vscode -> Visual Studio Code
What am I supposed to do if I download the package (as a user)?

Reply:
I was thinking we can have a wiki link with instructions on how to setup the packages manually and direct the users there. However in that case also their download might be corrupted. Or we can skip the last line and just say restart vscode. Thoughts ?

Reply:
I think for now I will just say "restart vscode" and if we people complain that they are getting these errors repeatedly we can write a wiki article that:

1. Helps them to identify their target OS
2. Identify which dependency are they missing
3. What is the link to download that dependency
4. How to set the appropriate permissions(on Linux/Mac)